TS2322 Typescript: 'keepAlive' type 'number' is not assignable to type 'never'

I have the following code:
```
public toClientConfig(): RedisOptions {
        return {
           host: this.host,
           port: this.port,
           db: this.db,
           password: this.password,
           keepAlive: 1,
           lazyConnect: true,
        };
     }
```

I got this error during the build:


```
src/cache/classes/cache-config.ts:27:10 - error TS2322: Type 'number' is not assignable to type 'never'.

27          keepAlive: 1,
            ~~~~~~~~~
node_modules/ioredis/built/redis/RedisOptions.d.ts:19:5
    19     keepAlive?: number;
           ~~~~~~~~~
    The expected type comes from property 'keepAlive' which is declared here on type 'RedisOptions'

```

I think it's a weird type overlap with **TcpSocketConnectOpts** 

```
interface TcpSocketConnectOpts extends ConnectOpts {
        port: number;
        host?: string | undefined;
        localAddress?: string | undefined;
        localPort?: number | undefined;
        hints?: number | undefined;
        family?: number | undefined;
        lookup?: LookupFunction | undefined;
        noDelay?: boolean | undefined;
        keepAlive?: boolean | undefined;
        keepAliveInitialDelay?: number | undefined;
    }
```


How can I solve it?

--- a/lib/connectors/StandaloneConnector.ts
+++ b/lib/connectors/StandaloneConnector.ts
@@ -4,8 +4,10 @@ import { NetStream } from "../types";
 import { CONNECTION_CLOSED_ERROR_MSG } from "../utils";
 import AbstractConnector, { ErrorEmitter } from "./AbstractConnector";
 
-export type StandaloneConnectionOptions = (Partial<TcpNetConnectOpts> &
-  Partial<IpcNetConnectOpts>) & {
+type TcpOptions = Pick<TcpNetConnectOpts, "port" | "host" | "family">;
+type IpcOptions = Pick<IpcNetConnectOpts, "path">;
+
+export type StandaloneConnectionOptions = Partial<TcpOptions & IpcOptions> & {
   disconnectTimeout?: number;
   tls?: ConnectionOptions;
 };
@@ -19,13 +21,13 @@ export default class StandaloneConnector extends AbstractConnector {
     const { options } = this;
     this.connecting = true;
 
-    let connectionOptions: TcpNetConnectOpts | IpcNetConnectOpts;
+    let connectionOptions: TcpOptions | IpcOptions;
     if ("path" in options && options.path) {
       connectionOptions = {
         path: options.path,
-      } as IpcNetConnectOpts;
+      } as IpcOptions;
     } else {
-      connectionOptions = {} as TcpNetConnectOpts;
+      connectionOptions = {} as TcpOptions;
       if ("port" in options && options.port != null) {
         connectionOptions.port = options.port;
       }

